functions/go/_template/README.md

```diff
@@ -6,20 +6,36 @@
 
 Explain what this function does in one or two sentences.
 
+Describe why the user should care about this function.
+
+What problem does it solve?
+
+Provide some context (e.g. In the `gatekeeper` function, explain what's
+is `Gatekeeper` project)
+
 [//]: <> (Note: The content between `<!--mdtogo:Short-->` and the following
 `<!--mdtogo-->` will be used as the short description for the command.)
 
 <!--mdtogo-->
 
+### Usage
+
+How do I use this function?
 
-### FunctionConfig
+Explain what does it do in details.
+
+Is this function meant to be used declaratively, imperatively or both?
+
+#### FunctionConfig
 
 <!--mdtogo:Long-->
 
-Explain the function config and behavior for this function in detail.
+Omit this section, if the function doesn't support any `functionConfigs`.
+Otherwise, explain the function config and behavior for this function in detail.
 For each field in the function config, specify:
- - An example value
- - Whether it is optional, and if so, the default value
+
+- An example value
+- Whether it is optional, and if so, the default value
 
 To decouple the function orchestrator (i.e. kpt) from the function, it's
```

Comment:
Is this still true? I think we are now embracing showing the CLI

Reply:
Updated